cockatrice/src/spoilerbackgroundupdater.cpp

```diff
@@ -0,0 +1,210 @@
+#include <QDateTime>
+#include <QDebug>
+#include <QUrl>
+#include <QNetworkReply>
+#include <QMessageBox>
+#include <QFile>
+#include <QApplication>
+#include <QtConcurrent>
+#include <QCryptographicHash>
+
+#include "spoilerbackgroundupdater.h"
+#include "settingscache.h"
+#include "carddatabase.h"
+#include "main.h"
+#include "window_main.h"
+
+#define SPOILERS_STATUS_URL "https://raw.githubusercontent.com/Cockatrice/Magic-Spoiler/files/SpoilerSeasonEnabled"
+#define SPOILERS_URL "https://raw.githubusercontent.com/Cockatrice/Magic-Spoiler/files/spoiler.xml"
+
+SpoilerBackgroundUpdater::SpoilerBackgroundUpdater(QObject *apParent) : QObject(apParent), cardUpdateProcess(nullptr)
+{
+    isSpoilerDownloadEnabled = settingsCache->getDownloadSpoilersStatus();
+    if (isSpoilerDownloadEnabled)
+    {
+        // Start the process of checking if we're in spoiler season
+        // File exists means we're in spoiler season
+        startSpoilerDownloadProcess(SPOILERS_STATUS_URL, false);
+    }
+}
+
+void SpoilerBackgroundUpdater::startSpoilerDownloadProcess(QString url, bool saveResults)
+{
+    auto spoilerURL = QUrl(url);
+    downloadFromURL(spoilerURL, saveResults);
+}
+
+void SpoilerBackgroundUpdater::downloadFromURL(QUrl url, bool saveResults)
+{
+    auto *nam = new QNetworkAccessManager(this);
+    QNetworkReply *reply = nam->get(QNetworkRequest(url));
+
+    if (saveResults)
+    {
+        // This will write out to the file (used for spoiler.xml)
+        connect(reply, SIGNAL(finished()), this, SLOT(actDownloadFinishedSpoilersFile()));
+    }
+    else
+    {
+        // This will check the status (used to see if we're in spoiler season or not)
+        connect(reply, SIGNAL(finished()), this, SLOT(actCheckIfSpoilerSeasonEnabled()));
+    }
+}
+
+void SpoilerBackgroundUpdater::actDownloadFinishedSpoilersFile()
+{
+    // Check for server reply
+    auto *reply = dynamic_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError errorCode = reply->error();
+
+    if (errorCode == QNetworkReply::NoError)
+    {
+        spoilerData = reply->readAll();
+        reply->deleteLater();
```

Comment:
Should this be done after `saveDownloadedFile`? If `readAll` doesn't transfer pointer ownership then I could see the save following this being a null issue, maybe

Reply:
I don't think it's necessary, but I'll appease you :)